esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,191 @@
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
+            if overwrite_lon_bounds or not cube_coord.has_bounds():
+                fix_bounds(cube, cube_coord)
+
+        # fix latitude
+        if cube_coord.var_name == 'lat':
+            logger.info("Fixing latitude...")
+            if overwrite_lat_bounds or not cube.coord('latitude').has_bounds():
+                fix_bounds(cube, cube.coord('latitude'))
+
+    return cube
+
+
+def extract_variable(var_info, raw_info, attrs, year):
+    """Extract variables."""
+    rawvar = raw_info['name']
+    constraint = iris.Constraint(name=rawvar)
+    if rawvar == 'sm_uncertainty':
+        sm_cube = iris.load_cube(raw_info['file'],
+                                 iris.NameConstraint(var_name='sm'))
+        ancillary_var = sm_cube.ancillary_variable(
+            'Volumetric Soil Moisture Uncertainty'
+        )
+        cube = sm_cube.copy(ancillary_var.core_data())
+    else:
+        cube = iris.load_cube(raw_info['file'], constraint)
+
+    # Continue with processing the cube if it was successfully loaded
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    fix_coords_esacci_soilmoisture(cube, overwrite_time_bounds=False)
+    set_global_atts(cube, attrs)
+
+    # Remove dysfunctional ancillary data without standard names
+    for ancillary_variable in cube.ancillary_variables():
+        cube.remove_ancillary_variable(ancillary_variable)
+
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg['attributes']
+
+    # run the cmorization
+    for var_name, vals in cfg['variables'].items():
+        all_data_cubes = []
+        if isinstance(vals, dict):  # Ensure vals is a dictionary
+            var_info = cfg['cmor_table'].get_variable(vals['mip'], var_name)
+            glob_attrs['mip'] = vals['mip']
+            raw_info = {'name': vals['raw']}
+            inpfile_pattern = os.path.join(in_dir, vals['filename'])
+            logger.info("CMORizing var %s from file type %s",
+                        var_name, inpfile_pattern)
+
+            for year in range(vals['start_year'], vals['end_year'] + 1):
+                year_inpfile_pattern = inpfile_pattern.format(year=year)
+                inpfiles = sorted(glob.glob(year_inpfile_pattern))
+                for inpfile in inpfiles:
+                    raw_info['file'] = inpfile
+                    logger.info("CMORizing var %s from file type %s",
+                                var_name, raw_info['file'])
+                    cube = extract_variable(var_info, raw_info, glob_attrs,
+                                            year)
+                    all_data_cubes.append(cube)
+            final_cube = concatenate(all_data_cubes)
+            time = final_cube.coord('time')
+            time.bounds = get_time_bounds(time, vals['frequency'])
```

Comment:
```suggestion
            final_cube.coord('time').guess_bounds()
```

is probably sufficient, and you can get rid of the `frequency` in the config file